File: cases/core/capability_safe/PY-CS-001/pr/base/agent/router.py
"""Agent router — dispatches plan steps to the appropriate tool."""

from __future__ import annotations

import logging

from agent.planner import PlanStep

logger = logging.getLogger(__name__)


def _skipped_result(command: str, reason: str) -> dict:
    return {
        "command": command,
        "allowed": False,
        "return_code": None,
        "stdout": "",
        "stderr": "",
        "errors": [reason],
    }


def execute_plan(plan: list[PlanStep]) -> list[dict]:
    """Execute each step in the coding-agent plan and collect results."""
    results: list[dict] = []

    for step in plan:
        if not isinstance(step, dict):
            logger.warning("Skipping non-dict plan step: %r", step)
            continue

        action = step.get("action")
        if not isinstance(action, str):
            results.append(_skipped_result("", "Plan step missing valid 'action' key"))
            continue

        # No tool implementations available yet — skip all actions.
        results.append(_skipped_result(
            step.get("command", ""),
            f"No handler registered for action: {action}",
        ))

    return results

File: cases/core/synthetic_vulnerable/RS-SV-002/pr/base/src/agent/router.rs
//! Agent router — dispatches build tasks to the appropriate tool.

use crate::agent::planner::BuildTask;
use crate::config::AgentConfig;

/// Result of executing a single build task.
#[derive(Debug)]
pub struct TaskResult {
    pub task_action: String,
    pub success: bool,
    pub output: String,
}

/// Execute each task in the build plan and collect results.
///
/// Build tool execution is not yet implemented; all tasks return an error.
pub fn execute_plan(plan: &[BuildTask], _config: &AgentConfig) -> Vec<TaskResult> {
    let mut results = Vec::new();

    for task in plan {
        let result = TaskResult {
            task_action: task.action.clone(),
            success: false,
            output: format!("Action '{}' is not yet implemented", task.action),
        };
        results.push(result);
    }

    results
}
